sound/soc/sof/compress.c

```diff
@@ -302,10 +302,21 @@ static int sof_compr_pointer(struct snd_soc_component *component,
 			     struct snd_compr_stream *cstream,
 			     struct snd_compr_tstamp *tstamp)
 {
+	uint64_t dai_posn;
+	struct snd_sof_pcm *spcm;
+	struct snd_soc_pcm_runtime *rtd = cstream->private_data;
 	struct sof_compr_stream *sstream = cstream->runtime->private_data;
 
+	spcm = snd_sof_find_spcm_dai(component, rtd);
+	if (!spcm)
+		return -EINVAL;
+
+	dai_posn = spcm->stream[cstream->direction].posn.dai_posn;
+
 	tstamp->sampling_rate = sstream->sampling_rate;
 	tstamp->copied_total = sstream->copied_total;
+	tstamp->pcm_io_frames = div_u64(spcm->stream[cstream->direction].posn.dai_posn,
+					sstream->channels * sstream->sample_container_bytes);
```

Comment:
Does this work if you have a mixer in your topology? I don't see how the status of the dai would be representative of the frame rendered?

Reply:
Another issue is that IIRC this timestamp was supposed to be usable on capture as well as playback. That's how I interpret the kernel-doc in compress_offload.h

````
 * @pcm_io_frames: Frames rendered or received by DSP into a mixer or an audio
 * output/input. 
````

The code above seems to be for playback only?

Reply:
Indeed we do not support compress capture yet. Should we add a comment now OR try to implement it for capture also? I don't think we even have a decoder library in the cadence package.

Reply:
@plbossart looking again at this, this should work for capture too. 

The number of processed frames (rendered or received) is kept by audio/dai-{legacy|zephyr}.c drivers. `total_data_processed` looks to be correctly updated in `dai_dma_cb` after a DMA copy has finished. 

We are yet to prove this as we are now adding support for Capture side of module adapter passthrough driver, @LaurentiuM1234  is working on.

Will discuss more about this at today's meeting.